generators/Options.fs

```diff
@@ -3,23 +3,42 @@ module Generators.Options
 open System
 open System.IO
 open CommandLine
+open CommandLine.Text
 
 type Status = 
     | Implemented
     | Unimplemented
     | MissingData
     | Deprecated
     | Custom
+    | Outdated
+    | All
 
-type CommandLineOptions = 
+type CommandLineOptions =
     { [<Option('e', "exercise", Required = false, 
-        HelpText = "Exercise to generate (if not specified, defaults to all exercises).")>] Exercise : string;
+        HelpText = "Single exercise to generate.")>] Exercise : string;
       [<Option('s', "status", Required = false, 
-        HelpText = "The generator status to filter on (defaults to exercises with generator).")>] Status : string;
+        HelpText = "List exercises with the given status (nothing is generated).")>] Status : string;
       [<Option('d', "canonicaldatadirectory", Required = false, 
         HelpText = "Canonical data directory. If the directory does not exist, the canonical data will be downloaded.")>] CanonicalDataDirectory : string;
       [<Option('c', "cachecanonicaldata", Required = false,
-        HelpText = "Use the cached canonical data and don't update the data.")>] CacheCanonicalData : bool; }
+        HelpText = "Use the cached canonical data and don't update the data.")>] CacheCanonicalData : bool; 
+    } with
+          static member Defaults () =
+              {
+                Exercise = null
+                Status = ""
+                CanonicalDataDirectory = null
+                CacheCanonicalData = false
+              }
+              
+          [<Usage(ApplicationAlias = "dotnet run")>]
+          static member Examples =
```

Comment:
This is great!

Reply:
This is my favorite feature of the CommandLine library. It's so slick to specify the usage programmatically, and get the string generated for you. Much less likely to rot the way manually-maintained documentation does.